packages/tpetra/core/src/Tpetra_Details_DistributorActor.hpp

```diff
@@ -278,11 +268,6 @@ void DistributorActor::doPosts(const DistributorPlan& plan,
     Teuchos::TimeMonitor timeMonBarrier (*timer_doPosts3KV_barrier_);
 #endif // HAVE_TPETRA_DISTRIBUTOR_TIMINGS
 
-    // If we are using ready sends (MPI_Rsend) below, we need to do
-    // a barrier before we post the ready sends.  This is because a
-    // ready send requires that its matching receive has already
-    // been posted before the send has been posted.  The only way to
-    // guarantee that in this case is to use a barrier.
     plan.getComm()->barrier ();
```

Comment:
oh I see what you mean now, and yeah, the comment must actually have been referring to that throw we used to have if barrierBetween wasn't true during readySend.  I don't know how to feel about doBarrier.  I don't love it, because I don't love having a bunch of production code that's only used to test things.  If it is easy to delete without rearranging a bunch of tests, I'm all for it.  My only concern would be if that messes up the testing logic in ways we don't understand.

Reply:
Thanks, @tasmith4 .  I will remove the doBarrier parameter option.  It was likely more useful during initial development of Distributor, and less useful now that Distributor is stable code.